Sources/DVR/Cassette.swift

```diff
@@ -18,16 +18,23 @@ struct Cassette {
 
     // MARK: - Functions
 
-    func interactionForRequest(_ request: URLRequest) -> Interaction? {
+    func interactionForRequest(_ request: URLRequest, requiredHeaders: [String] = []) -> Interaction? {
+        var match: Interaction?
         for interaction in interactions {
             let interactionRequest = interaction.request
 
-            // Note: We don't check headers right now
-            if interactionRequest.httpMethod == request.httpMethod && interactionRequest.url == request.url && interactionRequest.hasHTTPBodyEqualToThatOfRequest(request)  {
-                return interaction
+            if interactionRequest.httpMethod == request.httpMethod &&
+                interactionRequest.url == request.url &&
+                interactionRequest.hasHTTPBodyEqualToThatOfRequest(request)  {
+
+                // Overwrite the current match if the required headers are equal.
+                if match == nil ||
```

Comment:
Will `match` ever be non-nil? I don't see where it's set.

Reply:
It's set in the body of the condition if it's nil, and it will be overwritten if the "headersToCheck" have the same values.